packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -724,15 +724,35 @@ export class Project extends ProjectBase {
     const kmsIamResources = new Set<string>();
 
     for (const [name, envVariable] of Object.entries(environmentVariables)) {
-      const envVariableValue = envVariable.value?.toString();
+      let envVariableValue: string;
+      if (envVariable.value instanceof SecretValue && envVariable.value.secretQualifier) {
+        // validate that the plain-text environment variables don't contain any secrets in them
+        if (validateNoPlainTextSecrets && (!envVariable.type || envVariable.type === BuildEnvironmentVariableType.PLAINTEXT)) {
+          throw new Error(`Plaintext environment variable '${name}' contains a secret value! ` +
+            'This means the value of this variable will be visible in plain text in the AWS Console. ' +
+            "Please consider using CodeBuild's SecretsManager environment variables feature instead. " +
+            "If you'd like to continue with having this secret in the plaintext environment variables, " +
+            'please set the checkSecretsInPlainTextEnvVariables property to false');
+        }
```

Comment:
Sorry, what is happening here? This is all duplicated from the code below...?

All of this should be removed.

Are you maybe missing an additional condition from this line:

```ts
      if (envVariable.value instanceof SecretValue && envVariable.value.secretQualifier) {
```

that this logic should only apply if the type of the env variable is `SECRETS_MANAGER `?